.github/workflows/e2e.yml

```diff
@@ -25,10 +24,9 @@ jobs:
             target
           key: ${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.lock') }}-1
       - name: Install Rust
-        run: |
-          rustup update ${{ matrix.rust }} --no-self-update
-          rustup default ${{ matrix.rust }}
-          rustup component add rustfmt
+        uses: ructions/toolchain@v2
```

Comment:
Explicitly install Rust toolchain is not necessary anymore. GitHub actions runner has `rustup` by default. It will use the version/components in `rust-toolchain.toml` whenever call `cargo` commands.